controllers/apirule_controller.go

```diff
@@ -80,31 +109,36 @@ func (r *APIRuleReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		return r.setStatusForError(ctx, api, err, gatewayv1beta1.StatusSkipped)
 	}
 
-	//Prevent reconciliation after status update. It should be solved by controller-runtime implementation but still isn't.
-	if api.Generation != api.Status.ObservedGeneration {
+	validator := validation.APIRule{
+		ServiceBlockList:  r.ServiceBlockList,
+		DomainAllowList:   r.DomainAllowList,
+		HostBlockList:     r.HostBlockList,
+		DefaultDomainName: r.DefaultDomainName,
+	}
 
-		//1.1) Get the configuration
-		config, err := helpers.LoadConfig()
-		if err != nil {
-			//If configuration is not available not been able to continue.
-			r.Log.Info(fmt.Sprintf(`Configuration loading failed {"controller": "Api", "request": "%s/%s", "file": %s}`, api.Namespace, api.Name, helpers.CONFIG_FILE))
-			return r.setStatusForError(ctx, api, err, gatewayv1beta1.StatusError)
-		}
+	config, err := helpers.LoadConfig()
```

Comment:
We can check that the received req contains a ConfigMap, and then proceed to update if needed, instead of running the LoadConfig every loop

Reply:
Yes, we could optimize that, although small one.